docs/examples/use_cases/mxnet/mxnet-resnet50.ipynb

```diff
@@ -43,88 +44,74 @@
+    "def get_dali_pipeline(batch_size, num_threads, device_id, db_folder, crop, size,\n",
+    "                       shard_id, num_shards, dali_cpu=False, is_training=True):\n",
+    "    pipeline = Pipeline(batch_size, num_threads, device_id, seed=12 + device_id)\n",
+    "    with pipeline:\n",
+    "        images, labels = fn.mxnet_reader(path=[db_folder+\"train.rec\"], index_path=[db_folder+\"train.idx\"],\n",
+    "                                          random_shuffle=False, shard_id=device_id, num_shards=num_shards,\n",
+    "                                          pad_last_batch=is_training, name=\"Reader\")\n",
+    "        dali_device = 'cpu' if dali_cpu else 'gpu'\n",
+    "        decoder_device = 'cpu' if dali_cpu else 'mixed'\n",
+    "        device_memory_padding = 211025920 if decoder_device == 'mixed' else 0\n",
+    "        host_memory_padding = 140544512 if decoder_device == 'mixed' else 0\n",
+    "        if is_training:\n",
+    "            images = fn.image_decoder_random_crop(images,\n",
+    "                                                  device=decoder_device, output_type=types.RGB,\n",
+    "                                                  device_memory_padding=device_memory_padding,\n",
+    "                                                  host_memory_padding=host_memory_padding,\n",
+    "                                                  random_aspect_ratio=[0.8, 1.25],\n",
+    "                                                  random_area=[0.1, 1.0],\n",
+    "                                                  num_attempts=100)\n",
+    "            images = fn.resize(images,\n",
+    "                               device=dali_device,\n",
+    "                               resize_x=crop,\n",
+    "                               resize_y=crop,\n",
+    "                               interp_type=types.INTERP_TRIANGULAR)\n",
+    "            mirror = fn.coin_flip(probability=0.5)\n",
+    "        else:\n",
+    "            images = fn.image_decoder(images,\n",
+    "                                      device=decoder_device,\n",
+    "                                      output_type=types.RGB)\n",
+    "            # Make sure that every image > 224 for CropMirrorNormalize\n",
+    "            images = fn.resize(images,\n",
+    "                               device=dali_device,\n",
+    "                               resize_shorter=size,\n",
+    "                               interp_type=types.INTERP_TRIANGULAR)\n",
+    "            mirror = False\n",
     "\n",
-    "    def define_graph(self):\n",
-    "        self.jpegs, self.labels = self.input(name = \"Reader\")\n",
-    "        images = self.decode(self.jpegs)\n",
-    "        output = self.cmnp(images)\n",
-    "        return [output, self.labels]\n"
+    "        images = fn.crop_mirror_normalize(images.gpu(),\n",
+    "                                          device=\"gpu\",\n",
```

Comment:
You don't have to specify `device="gpu"` - it's inferred from the input and in this case, clearly visible (`images.gpu()`)
```suggestion
```

Reply:
Done